eng/testing/debug-dump-template.md

```diff
@@ -0,0 +1,122 @@
+
+### How do I install it?
+
+You just need to run:
+
+```script
+dotnet tool install --global runfo
+```
+
+If you already have it installed, make sure you have at least version `0.6.1` installed, which contains support to download helix payloads. If you don't have the latest version just run:
+
+```script
+dotnet tool update --global runfo
+```
+
+## Download helix payload containing symbols:
+
+You can just achieve this by running:
+
+```script
+runfo get-helix-payload -j %JOBID% -w %WORKITEM% -o <out-dir>
+```
+
+> NOTE: if the helix job is an internal job, you need to pass down a [helix authentication token](https://helix.dot.net/Account/Tokens) using the `--helix-token` argument.
+
+This will download the workitem contents under `<out-dir>\workitems\` and the correlation payload under: `<out-dir>\correlation-payload\`. 
+
+> The correlation payload is usually the testhost or core root, which contain the runtime and dotnet host that we use to run tests.
+
+Once you have those assets, you will need to extract the testhost or core root. Then extract the workitem assets into the same location where coreclr binary is.
+
+## Windows dump on windows
+
+### Debug with WinDbg
+
+1. Install [dotnet-sos global tool](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-sos).
+2. Load the dump with a recent WinDbg.
+3. Then run the following commands:
+
+```script
+!setclrpath <path to core root or testhost where coreclr is>
+!sethostruntime '<path to your local dotnet runtime or testhost>'
+.sympath+ <directory with symbols (for library tests these are in testhost dir)>
+```
+### Analyze with dotnet-dump
+
+1. Install [dotnet-dump global tool](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-dump).
+2. Run: `dotnet-dump analyze <path-to-dump>`
+3. Then run the following commands:
+
+```script
+setclrpath <you should see `Load path for DAC/DBI: '<none>'`>
+setclrpath <path to core root or testhost where coreclr is>
+setsymbolserver -directory <directory with symbols (for library tests these are in testhost dir)>
+```
+
+## Linux dumps on windows
+
+In order to debug a Linux dump on Windows, you will have to first go to the PR/CI build
+that sent the test run and download the cross DAC.
+
+Go to your build by navigating to [your build](https://dnceng.visualstudio.com/public/_build/results?buildId=%BUILDID%&view=artifacts&type=publishedArtifacts) and download the `CoreCLRCrossDacArtifacts`, then choose the cross DAC you need depending on the flavor.
+
+Once you've downloaded that artifact, make sure to extract it in the same location where coreclr binary is.
+
+### Debug with WinDbg
+
+1. Install [dotnet-sos global tool](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-sos).
+2. Load the dump with a recent WinDbg.
+3. Then run the following commands:
+
+```script
+.load '<path to sos (can get it by running `dotnet-sos install`)>'
```

Comment:
If you go with the manual load route, this line should also be in the above script for Windows dumps.

Reply:
And make sure to unload the one that might be bundled first. This is only necessary if the version of SOS in the gallery doesn't have the needed commands and bug fixes yet. This is probably not all too common.

Reply:
When opening the Linux dump it didn't automatically load the sos plugin for me, but when opening a Windows dump it did... do we know why? I'll have to try this again.

Reply:
On linux, did you run `dotnet sos install`?

Reply:
I think he means linux dumps under windbg and I think there is a bug that I need to follow up on that sos doesn't get loaded for linux dumps (triggered by the libcoreclr.so module). The command `.update sos` should load it. I need to follow up with the windbg about why it isn't automatically being loaded.

Reply:
I have version 10.0.20212.1001 and it loads sos.dll for my linux dumps now. Maybe you need a newer windbg version.

Reply:
> I have version 10.0.20212.1001 and it loads sos.dll for my linux dumps now. Maybe you need a newer windbg version.

Ah that's it. I have a quite old version. Does it make sense mentioning `.update sos` command just in case?

Reply:
You can mention it, but it would be better to make everybody has a new version.

Reply:
Sounds good.